### PR TITLE
Test monitor GitHub actions

### PR DESCRIPTION
**Pull request by smnzhu**
TODO:
create a reusable workflow: https://github.com/onflow/flow-go/issues/2213

closes https://github.com/onflow/flow-go/issues/2163

### CODE REVIEW

File: utils/unittest/unittest.go
Comment (gomisha) [inline utils/unittest/unittest.go:52]: can you add some explanation of what's happening here? We're checking the skipped test list to skip all the tests on that list?

File: tools/flaky_test_monitor/testdata/summary3/test3-multi-no-result-tests/input/test3-multi-no-result-tests.json
Comment (gomisha) [inline tools/flaky_test_monitor/testdata/summary3/test3-multi-no-result-tests/input/test3-multi-no-result-tests.json:1]: why is this test file completely deleted?

Comment (gomisha) [reply tools/flaky_test_monitor/testdata/summary3/test3-multi-no-result-tests/input/test3-multi-no-result-tests.json:1]: I see you renamed it to tools/flaky_test_monitor/testdata/summary3/test3-multi-exceptions/input/test3-multi-exceptions.json?

File: utils/unittest/unittest.go
Comment (smnzhu) [reply utils/unittest/unittest.go]: Sorry, that's outdated. I've updated the PR to use a different method for tracking skipped tests

File: .github/workflows/flaky-test-monitor.yml
Comment (gomisha) [inline .github/workflows/flaky-test-monitor.yml:9]: why do we run on every PR?

File: .github/workflows/flaky-test-monitor.yml
Comment (gomisha) [inline .github/workflows/flaky-test-monitor.yml:17]: why would we change the table before merging? I thought we would publish to the dev table for now

File: .github/workflows/flaky-test-monitor.yml
Comment (smnzhu) [reply .github/workflows/flaky-test-monitor.yml:9]: As the comment says, I'm gonna remove this before merging.

It's just there now so that every time I push, the jobs are triggered on this PR. It provides a convenient way to test my changes.

File: .github/workflows/skipped-test-tracker.yml
Comment (gomisha) [inline .github/workflows/skipped-test-tracker.yml:1]: can you add an explanation / comment to this job? Is it tracking which tests are skipped? If so, can you elaborate what the bigger purpose is?

File: tools/flaky_test_monitor/common/testdata/test_data.go
Comment (gomisha) [inline tools/flaky_test_monitor/common/testdata/test_data.go:33]: what's the purpose of changing the time stamp? to make it more realistic where there will be an offset from UTC?

File: tools/flaky_test_monitor/level1/process_summary1_results.go
Comment (gomisha) [inline tools/flaky_test_monitor/level1/process_summary1_results.go:159]: so we still want to collect skipped tests?

File: tools/flaky_test_monitor/level1/process_summary1_results.go
Comment (gomisha) [inline tools/flaky_test_monitor/level1/process_summary1_results.go:130]: this comment is out of date since we're not removing skipped tests anymore

File: tools/flaky_test_monitor/level1/process_summary1_results.go
Comment (gomisha) [inline tools/flaky_test_monitor/level1/process_summary1_results.go:148]: can you add a comment on why we're starting not from the last output element? Is this assuming that the last output was a pass?

File: tools/flaky_test_monitor/level1/process_summary1_results.go
Comment (gomisha) [inline tools/flaky_test_monitor/level1/process_summary1_results.go:149]: can you add a comment on why we're starting not from the last output element? Is this assuming that the last output was a pass?

File: tools/flaky_test_monitor/level1/process_summary1_results_test.go
Comment (gomisha) [inline tools/flaky_test_monitor/level1/process_summary1_results_test.go:108]: Why were the JSON tests removed? I thought we agreed to having a minimal number of JSON tests (while the rest are struct tests)

File: tools/flaky_test_monitor/testdata/summary2/test3-multi-no-result-tests/expected-output/test3-multi-no-result-tests.json
Comment (gomisha) [inline tools/flaky_test_monitor/testdata/summary2/test3-multi-no-result-tests/expected-output/test3-multi-no-result-tests.json:1]: was this deleted file replaced? I can see you changed the names of the `no_result` field to `exception` but I don't see a replacement for this file

File: tools/flaky_test_monitor/common/testdata/test_data.go
Comment (gomisha) [inline tools/flaky_test_monitor/common/testdata/test_data.go:443]: nit: comment can be updated to use `tests with exceptions` instead of `no result tests`

File: tools/flaky_test_monitor/common/testdata/test_data.go
Comment (gomisha) [inline tools/flaky_test_monitor/common/testdata/test_data.go:267]: nit: var name shouldn't be capitlized

File: tools/flaky_test_monitor/common/testdata/test_data.go
Comment (gomisha) [inline tools/flaky_test_monitor/common/testdata/test_data.go:245]: nit: var name shouldn't be capitalized

File: tools/flaky_test_monitor/common/testdata/test_data.go
Comment (gomisha) [inline tools/flaky_test_monitor/common/testdata/test_data.go:93]: why is the expected output removed? it's passed as a parameter and should be part of the generated level 1 test result

File: tools/flaky_test_monitor/common/testdata/test_data.go
Comment (gomisha) [inline tools/flaky_test_monitor/common/testdata/test_data.go:63]: why is the expected output removed?

File: tools/flaky_test_monitor/common/level1.go
Comment (gomisha) [inline tools/flaky_test_monitor/common/level1.go:48]: I like this change. So we don't need to wrap all the level 1 test results inside a struct but just treat the list of results as a separate type instead

File: tools/flaky_test_monitor/common/level2.go
Comment (gomisha) [inline tools/flaky_test_monitor/common/level2.go:31]: this looks like a utility function that doesn't belong here. Maybe into `utils/unittest.go` or `tools/flaky_test_monitor/common/utility.go`

File: tools/flaky_test_monitor/level1/process_summary1_results.go
Comment (smnzhu) [reply tools/flaky_test_monitor/level1/process_summary1_results.go:149]: Sure, I'll add a comment. It's just because in most cases the line containing the skip reason is the last line of output (but not always).

File: tools/flaky_test_monitor/level1/process_summary1_results_test.go
Comment (smnzhu) [reply tools/flaky_test_monitor/level1/process_summary1_results_test.go:108]: Yeah I can probably add them back. There were a bunch of changes that had to be made so I just deleted them to unblock myself.

File: tools/flaky_test_monitor/common/testdata/test_data.go
Comment (smnzhu) [reply tools/flaky_test_monitor/common/testdata/test_data.go:59]: See https://github.com/onflow/flow-go/blob/95af2ab407bb46c38f85a6e52e4f9afe32e3d323/tools/flaky_test_monitor/level1/process_summary1_results.go#L202

File: .github/workflows/flaky-test-monitor-regression.yml
Comment (smnzhu) [reply .github/workflows/flaky-test-monitor-regression.yml:1]: Sure will do. But to answer your question, it will be run on every PR or on bors branches (staging and trying). This is necessary because the regular flaky test monitor job will actually push data to bigquery, but we don't want to do this for the test.

**Review by gomisha — APPROVED** [review]

File: consensus/hotstuff/committees/consensus_committee_test.go
Comment (jordanschalm) [inline consensus/hotstuff/committees/consensus_committee_test.go:142]:
```suggestion
			unittest.SkipUnless(t, unittest. TEST_WIP, "disabled as the current implementation uses a temporary fallback measure in this case (triggers EECC), rather than returning an error")
```

File: integration/tests/collection/proposal.go
Comment (jordanschalm) [inline integration/tests/collection/proposal.go:232]:
```suggestion
	unittest.SkipUnless(t, unittest.TEST_WIP, "incomplete implementation - disconnection of one node by disconnecting from Docker network is unreliable")
```

File: state/protocol/badger/mutator_test.go
Comment (jordanschalm) [inline state/protocol/badger/mutator_test.go:1148]:
```suggestion
	unittest.SkipUnless(t, unittest.TEST_WIP, "disabled as the current implementation uses a temporary fallback measure in this case (triggers EECC), rather than returning an error")
```

**Review by jordanschalm — COMMENTED** [review]
Adding some skip reasons

File: cmd/access/node_builder/staked_access_node_builder.go
Comment (Kay-Zee) [inline cmd/access/node_builder/staked_access_node_builder.go:325]: I guess this is the old version, worth merging the new version in?

File: module/metrics/herocache.go
Comment (Kay-Zee) [inline module/metrics/herocache.go:33]: here as well, i think we had a var to represent this, rather than the string concat

File: tools/flaky_test_monitor/level1/process_summary1_results.go
Comment (Kay-Zee) [reply tools/flaky_test_monitor/level1/process_summary1_results.go:151]: wait, but isn't this starting from the second last line?

File: tools/flaky_test_monitor/level1/process_summary1_results_test.go
Comment (Kay-Zee) [reply tools/flaky_test_monitor/level1/process_summary1_results_test.go:108]: Which tests did we decide to keep?

**Review by Kay-Zee — APPROVED** [review]
Some minor comments

File: tools/flaky_test_monitor/level1/process_summary1_results.go
Comment (smnzhu) [reply tools/flaky_test_monitor/level1/process_summary1_results.go:151]: Yes, the actual last line is output by go. The second last line is the last user outputted line

File: tools/flaky_test_monitor/level1/process_summary1_results_test.go
Comment (smnzhu) [reply tools/flaky_test_monitor/level1/process_summary1_results_test.go:108]: All the rest.

Comment (smnzhu) [reply tools/flaky_test_monitor/level1/process_summary1_results_test.go:108]: They can be added back in a followup PR if desired

**Comment by smnzhu** [comment]
bors merge

**Comment by smnzhu** [comment]
bors merge

**Pull request merged**
